src/retrieve-validation/process-parameters.ts

```diff
@@ -0,0 +1,18 @@
+// Imports
+import retrieveValidation from '.';
+import buildRequestURLParameters from './build-request-url-parameters';
+import { Parameters as RequestParameters } from '../types/parameters';
+
+// Function that processes parameters into the correct data type / structure based on the request method
+const processParameters = (method: Parameters<typeof retrieveValidation>[0], parameters: RequestParameters): string => {
+	// Handle parameters for GET method
+	if (method === 'GET') {
+		return buildRequestURLParameters(parameters);
+	}
+
+	// Throw if an unrecognized parameter is provided
+	// eslint-disable-next-line @typescript-eslint/restrict-template-expressions -- This should be unreachable
+	throw new Error(`Parameter processing called with unrecognized method: ${method}`);
```

Comment:
Would this prevent you from having to disable this ESLint rule?

```suggestion
	throw new Error(`Parameter processing called with unrecognized method: ${method ?? 'UNKNOWN'}`);
```

Reply:
Yes.

Reply:
Actually it doesn't the problem wasn't that method was nullish, but just rather that this should be unreachable so `method` is of type `never` here.